csvcubed/tests/unit/models/test_unit.py

```diff
@@ -65,6 +65,28 @@ def test_known_existing_units_defined():
     assert_num_validation_errors(errors, 0)
 
 
+def test_known_existing_units_defined_non_standard_uris():
+    """Ensure that we don't get any errors raised when existing units are used without
+    standard uri conventions."""
+    data = pd.DataFrame({"Unit": ["NUM", "PERCENT"]})
+    unit_column = QbColumn(
+        "Unit",
+        QbMultiUnits(
+            [
+                ExistingQbUnit("http://example.org/units/NUM"),
+                ExistingQbUnit("http://example.org/units/PERCENT"),
+            ]
+        ),
+    )
+
+    unit_dimension = unit_column.structural_definition
```

Comment:
It's not a dimension, it's just some kind of DSD definition which lists the units used in the dataset.

And I'd suggest the following approach would be a bit simpler and closer to the behaviour I'd expect to happen:

```python
    data = pd.DataFrame({"Unit": ["NUM", "PERCENT"]})
    unit_column = QbColumn(
        "Unit",
        QbMultiUnits(
            [
                ExistingQbUnit("http://example.org/units/NUM"),
                ExistingQbUnit("http://example.org/units/PERCENT"),
            ]
        ),
        csv_column_uri_template="http://example.org/units/{+unit}"
    )
    
    errors = unit_column.validate_data(data["Unit"])

    assert_num_validation_errors(errors, 0)
```